pytest/tests/loadtest/locust/common/base.py

```diff
@@ -0,0 +1,324 @@
+import base64
+import json
+import base58
+import ctypes
+import locust
+import logging
+import multiprocessing
+import pathlib
+import random
+import requests
+import sys
+import time
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[4] / 'lib'))
+
+import cluster
+import transaction
+import key
+import mocknet_helpers
+import utils
+
+from locust import HttpUser, events
+from configured_logger import new_logger
+
+DEFAULT_TRANSACTION_TTL_SECONDS = 20
+logger = new_logger(level=logging.WARN)
+
+
+class Account:
+
+    def __init__(self, key):
+        self.key = key
+        self.current_nonce = multiprocessing.Value(ctypes.c_ulong, 0)
+
+    def refresh_nonce(self, node):
+        with self.current_nonce.get_lock():
+            self.current_nonce.value = mocknet_helpers.get_nonce_for_key(
+                self.key,
+                addr=node.rpc_addr()[0],
+                port=node.rpc_addr()[1],
+                logger=logger,
+            )
+
+    def use_nonce(self):
+        with self.current_nonce.get_lock():
+            new_nonce = self.current_nonce.value + 1
+            self.current_nonce.value = new_nonce
+            return new_nonce
+
+
+class Transaction:
+    """
+    A transaction future.
+    """
+
+    ID = 0
+
+    def __init__(self):
+        self.id = Transaction.ID
+        Transaction.ID += 1
+
+        # The transaction id hash
+        #
+        # str if the transaction has been submitted and may eventually conclude.
+        # FIXME: this is currently not set in some cases
+        self.transaction_id = None
+
+    def sign_and_serialize(self, block_hash):
+        """
+        Each transaction class is supposed to define this method to serialize and
+        sign the transaction and return the raw message to be sent.
+        """
+        return None
+
+
+class Deploy(Transaction):
+
+    def __init__(self, account, contract, name):
+        super().__init__()
+        self.account = account
+        self.contract = contract
+        self.name = name
+
+    def sign_and_serialize(self, block_hash):
+        account = self.account
+        logger.info(f"deploying {self.name} to {account.key.account_id}")
```

Comment:
nit: this looks like it's logged already when it's sent in `pytest/tests/loadtest/locust/locustfile.py`, so there will be two log lines for each Deploy tx sent. Could just log it with the account and key info once when sending it

Reply:
right, let me remove the logs in `pytest/tests/loadtest/locust/locustfile.py`